Added an explanation for the definition of a functor (if helpful); ad… (#55)
benediktahrens/CT4P: tex/CT4P.tex

\end{itemize}
\end{dfn}

\begin{rem}[Explanation]
	A function from a set $X$ to a set $Y$ maps every element of $X$, to an element in $Y$.

> 	Analogously, a \textit{functor} from a category $\CC$ to a category $\DD$ maps every objects of $\CC$, to an object in $\DD$.

Reviewer: ```suggestion
	Analogously, a \emph{functor} from a category $\CC$ to a category $\DD$ maps every object of $\CC$ to an object in $\DD$.
```

	(This is the first piece of data in the definition of a functor; made formal as a function between the underlying types/collections/sets of objects.)

> 	However, a category does not only consists of a set of objects.

Reviewer: ```suggestion
	However, a category does not only consist of a set of objects.
```

	(Indeed, between every two objects $c_1, c_2 \in \Ob{\CC}$, we have a set of morphisms; denoted $\CHom{\CC}{x}{y}$).

> 	Hence, a functor, as a transformation between categories, should also map every morphism in $\CC$, to a (suitably typed) morphism in $\DD$.

Reviewer: ```suggestion
	Hence, a functor, as a ``morphism'' between categories, should also map every morphism in $\CC$ to a morphism in $\DD$ (while suitably preserving domain and codomain of the morphism).
```

	(This is the second piece of data in the definition of a functor.)
	
	In the examples and exercises below, the first piece (i.\,e.,\, the function between the collection of objects) is always given.

> 	(Why? Because between fixed categories, there may be many, or none, functors.)

Reviewer: ```suggestion
	(Why? Because between fixed categories, there may be many functors, or none.)
```

	While the second piece of data is not uniquely determined by the function between the object-sets, there is often a canonical (obvious) choice.
	Below, we ask the reader to find the canonical choice.
	A functor is completely determined by the function (or action) on objects and morphisms.
	However, in order to guarantee that these functions are well-behaved (w.\,r.\,t.,\, to the structure given by the categories), we furthermore require that the composition and identity are preserved.

> 	These properties are not defined by the user. Nonetheless, they remain to be checked.

Reviewer: ```suggestion
They need to be checked to complete the definition of a functor.
```
